src/util/simplify_expr2.cpp

```diff
@@ -852,6 +852,17 @@ expr2tc with2t::do_simplify() const
 
 expr2tc member2t::do_simplify() const
 {
+  expr2tc current_source = source_value;
+  while(is_with2t(current_source))
+  {
+    const with2t &with = to_with2t(current_source);
+    if(member == to_constant_string2t(with.update_field).value)
```

Comment:
I am surprised that this works. The `with2t::update_field` (unfortunately) can also be an integer-typed expression in case the `source_value` is an array. Could this be a reason for the slow-down you see?

Reply:
How are you measuring "performance down"? What's the `CMAKE_BUILD_TYPE` you're using for that?

Reply:
> I am surprised that this works. The `with2t::update_field` (unfortunately) can also be an integer-typed expression in case the `source_value` is an array. Could this be a reason for the slow-down you see?

Do you mean the struct like with this structure?
```
struct stru{
    int arr[2];
};
```
And also we need to const propag for array:
```
int arr[2];
arr[0] = 1;
fun_array(arr);
```